pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -320,7 +320,25 @@ func (ctrl *PersistentVolumeController) syncUnboundClaim(claim *v1.PersistentVol
 			// OBSERVATION: pvc is "Pending", will retry
 			switch {
 			case delayBinding && !pvutil.IsDelayBindingProvisioning(claim):
-				ctrl.eventRecorder.Event(claim, v1.EventTypeNormal, events.WaitForFirstConsumer, "waiting for first consumer to be created before binding")
+				reason := events.WaitForFirstConsumer
+				message := "waiting for first consumer to be created before binding"
+				podNames, err := ctrl.findNonScheduledPodsByPVC(claim)
+				if err != nil {
+					return err
+				}
+				if len(podNames) > 0 {
+					reason = events.BindingWithPod
+					if len(podNames) > 1 {
+						// Although only one pod is taken into account in
+						// volume scheduling, more than one pods can reference
+						// the PVC at the same time. We can't know which pod is
+						// used in scheduling, all pods are included.
+						message = fmt.Sprintf("binding with pods %s", strings.Join(podNames, ","))
```

Comment:
The message is not clear to me. User sees this:
```
waiting for first consumer to be created before binding
binding with the pod pod-0
```
And the PVC is still unbound. What is the PVC waiting for?

IMO, something like this would be more informative: "waiting for pod X to be scheduled".

Reply:
it makes sense

Reply:
I've also renamed the reason `BindingWithPod` to 'WaitingForPodScheduled`. The pod may be stuck in the scheduling queue for some reason. It's more correct to say the PVC is waiting for the Pod to be scheduled.